strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/maven/MavenIndexController.java

```diff
@@ -1,103 +1,149 @@
 import org.carlspring.strongbox.providers.io.RepositoryPath;
-import org.carlspring.strongbox.providers.layout.LayoutProviderRegistry;
-import org.carlspring.strongbox.services.ArtifactIndexesService;
-import org.carlspring.strongbox.storage.ArtifactStorageException;
-import org.carlspring.strongbox.storage.Storage;
-import org.carlspring.strongbox.storage.indexing.IndexTypeEnum;
+import org.carlspring.strongbox.repository.MavenRepositoryFeatures;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexFounder;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexFounder.RepositoryIndexFounderQualifier;
 import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryTypeEnum;
+import org.carlspring.strongbox.web.RepositoryMapping;
 
 import javax.inject.Inject;
-import javax.ws.rs.QueryParam;
 import java.io.IOException;
 
-import io.swagger.annotations.*;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.context.annotation.Conditional;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
-import static org.carlspring.strongbox.util.IndexContextHelper.getContextId;
 
 /**
  * @author Kate Novik
  * @author carlspring
+ * @author Przemyslaw Fusik
  */
 @RestController
-@Api(value = "/api/maven/index")
-@Conditional(MavenIndexerEnabledCondition.class)
+@RequestMapping("/api/maven/index")
 public class MavenIndexController
         extends BaseController
 {
 
     private static final Logger logger = LoggerFactory.getLogger(MavenIndexController.class);
 
     @Inject
-    private ArtifactIndexesService artifactIndexesService;
-    
+    private MavenRepositoryFeatures mavenRepositoryFeatures;
+
+    @Inject
+    @RepositoryIndexFounderQualifier(RepositoryTypeEnum.HOSTED)
+    private RepositoryIndexFounder hostedRepositoryIndexFounder;
+
     @Inject
-    private LayoutProviderRegistry layoutProviderRegistry;
+    @RepositoryIndexFounderQualifier(RepositoryTypeEnum.PROXY)
+    private RepositoryIndexFounder proxyRepositoryIndexFounder;
 
+    @Inject
+    @RepositoryIndexFounderQualifier(RepositoryTypeEnum.GROUP)
+    private RepositoryIndexFounder groupRepositoryIndexFounder;
+
+    @PreAuthorize("hasAuthority('MANAGEMENT_REBUILD_INDEXES')")
+    @PostMapping(value = "/hosted/{storageId}/{repositoryId}", produces = { MediaType.TEXT_PLAIN_VALUE,
+                                                                            MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity rebuildHostedRepositoryIndex(@RepositoryMapping Repository repository)
```

Comment:
@fuss86 we can create the `MavenFileSystem.rebuildIndex` method and in this case there will be no need for three separate endpoints here.

Reply:
Good catch! I agree, there should be one end-point and the controller should be able to figure things out and delegate to the respective service methods.

Reply:
done